tests/components/caldav/test_calendar.py

```diff
@@ -1140,3 +1162,104 @@ async def test_config_entry_supported_components(
     # No entity created when no components exist
     state = hass.states.get("calendar.calendar_4")
     assert not state
+
+
+@pytest.mark.parametrize("tz", [UTC])
+@pytest.mark.parametrize(
+    ("service_data", "expected_ics_fields"),
+    [
+        # Basic event with all fields
+        (
+            {
+                "summary": "Test Event",
+                "start_date_time": datetime.datetime(2025, 8, 6, 10, 0, 0),
```

Comment:
I think it's more accurate for the input service call to take a string rather than a date object